squirrel.frontier/src/main/java/org/dice_research/squirrel/data/uri/norm/UriGeneratorImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.dice_research.squirrel.data.uri.norm;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.dice_research.squirrel.data.uri.CrawleableUri;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class UriGeneratorImpl implements UriGenerator {
```

Comment:
Could you please separate the two functionalities that you have implemented in this class? E.g., a `DomainBasedUriGenerator` and a `WellKnownPathUriGenerator` would be better.